src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -57,43 +60,43 @@ abstract class ListDatabaseRequestHandler<T> : IListDatabaseRequestHandler
     {
         private static readonly string[] SystemDatabases = new string[] { "master", "model", "msdb", "tempdb" };
 
-        public abstract string QueryText { get; }
-
         public ListDatabasesResponse HandleRequest(ISqlConnectionFactory connectionFactory, ConnectionInfo connectionInfo)
         {
             ConnectionDetails connectionDetails = connectionInfo.ConnectionDetails.Clone();
-
-            // Connect to master
-            connectionDetails.DatabaseName = "master";
-            using (var connection = connectionFactory.CreateSqlConnection(ConnectionService.BuildConnectionString(connectionDetails), connectionDetails.AzureAccountToken))
+            var connection = connectionFactory.CreateSqlConnection(ConnectionService.BuildConnectionString(connectionDetails), connectionDetails.AzureAccountToken);
+            
+			var results = new List<T>();
+            ListDatabasesResponse response = new ListDatabasesResponse();
+			
+            ServerConnection serverConnection = new ServerConnection(connection as SqlConnection);
+            try
             {
-                connection.Open();
-                ListDatabasesResponse response = new ListDatabasesResponse();
-                using (DbCommand command = connection.CreateCommand())
+                SMO.Server server = new SMO.Server(serverConnection);
+
+                // Using this function to set the default init properties of the server so that those things can be fetched in a single query.
+                server.SetDefaultInitFields(typeof(Database), DefaultServerInitProperties);
+
+                foreach (SMO.Database db in server.Databases)
```

Comment:
Currently you're changing the behavior here - previously we were only listing databases where state was ONLINE. You should add that back in unless we decide we want to display offline DBs as well.